WordPress/Classes/ViewRelated/Post/Prepublishing/PrepublishingViewController+JetpackSocial.swift

```diff
@@ -197,9 +169,72 @@ private extension PrepublishingViewController {
         }
     }
 
+    // MARK: - Model Creation
+
+    func makeAutoSharingModel() -> PrepublishingAutoSharingModel {
+        return coreDataStack.performQuery { [postObjectID = post.objectID] context in
+            guard let post = (try? context.existingObject(with: postObjectID)) as? Post,
+                  let connections = post.blog.sortedConnections as? [PublicizeConnection],
+                  let supportedServices = try? PublicizeService.allSupportedServices(in: context) else {
+                return .init(services: [], message: String(), sharingLimit: nil)
+            }
+
+            // first, build a dictionary to categorize the connections.
+            var connectionsMap = [PublicizeService.ServiceName: [PublicizeConnection]]()
+            connections.filter { !$0.requiresUserAction() }.forEach { connection in
```

Comment:
I'm filtering out invalid/broken connections here to keep things simple for now.